### DISCUSSION TITLE
runc dont kill container

### DISCUSSION THREAD

**Discussion by lengrongfu** [opencontainers/runc#3752] [Q&A]
Reproduce steps：

- 1. create `rootfs` file
```shell
# create the top most bundle directory
$ mkdir /mycontainer
$ cd /mycontainer

# create the rootfs directory
$ mkdir rootfs

# export busybox via Docker into the rootfs directory
$ docker export $(docker create busybox) | tar -C rootfs -xvf -
```

- 2. create spec 
```shell
$ cd mycontainer 
$ runc spec
```

- 3. update config.json
```sh
# Do not modify other
"terminal": true, => "terminal": false,
```

- 4. create and start container
```shell
$ runc create test
$ runc start test
```
- 5. kill test container
```shell
$ runc kill test # but kill failed
```

**Comment by fahedouch** [comment]
Hi @lengrongfu, 

it depends on the `init` process of you container (default `sh`). The `init` process is a special process. It behaves differently then normal process when receiving unix signals. 

the above command will send a `SIGTERM` for you container init process
`runc kill test # but kill failed`

So you have to check how your init process handles the SIGTERM, otherwise you send more powerful sig e.g ( `runc kill test SIGKILL`

I recommend you to [this reference](https://raby.sh/sigterm-and-pid-1-why-does-a-container-linger-after-receiving-a-sigterm.html) to help you debug your issue

**Comment by lengrongfu** [comment]
I am just doing a test, do we need special treatment for `sh` as a container process, to ensure that the container status can change to `stopped` after the user executes kill, and then `runc delete` can be executed normally.

**Comment by kolyshkin** [comment]
As described by @fahedouch above, all that `runc kill` does is sending a signal to the container's init process (which, in this case, is busybox shell). The default signal is `SIGTERM`, and how a process is handling it depends on the process.

If you do need to forcibly kill and delete a container, use `runc delete -f <container-id>` or `runc kill <container-id> KILL` followed by `runc delete <container-id>`.

I don't see how what you describe is a bug, thus moving to discussion

**Comment by lengrongfu** [comment]
I see, thank you.
